app/code/Magento/Sales/Block/Adminhtml/Report/Filter/Form/Coupon.php

```diff
@@ -3,17 +3,28 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Sales\Block\Adminhtml\Report\Filter\Form;
 
+use Magento\Backend\Block\Template\Context;
+use Magento\Framework\Data\Form\Element\Fieldset;
+use Magento\Framework\Data\FormFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Registry;
+use Magento\Reports\Block\Adminhtml\Filter\Form as ReportsBlockFilterForm;
+use Magento\Sales\Model\Order\ConfigFactory;
+use Magento\SalesRule\Model\ResourceModel\Report\RuleFactory;
+use Magento\Sales\Block\Adminhtml\Report\Filter\Form as SalesBlockForm;
+
 /**
  * Sales Adminhtml report filter form for coupons report
  *
  * @api
  * @author     Magento Core Team <core@magentocommerce.com>
  * @SuppressWarnings(PHPMD.DepthOfInheritance)
- * @since 100.0.2
+ * @since      100.0.2
  */
-class Coupon extends \Magento\Sales\Block\Adminhtml\Report\Filter\Form
+class Coupon extends SalesBlockForm
```

Comment:
- I added new $itemSortLimit construct parameter to Magento\Sales\Block\Adminhtml\Report\Filter\Form. So its options are initialized as arguments in di.xml.
- Since \Magento\Sales\Block\Adminhtml\Report\Filter\Form\Coupon is extended from it, I've changed parent construct call by adding an empty array to the end - parent::__construct($context, $registry, $formFactory, $orderConfig, $data, []);
- And some refactoring was done as well.